etl-api/src/routes/pipelines.rs

```diff
@@ -340,12 +335,52 @@ pub struct TableReplicationStatus {
     #[schema(example = "public.users")]
     pub table_name: String,
     pub state: SimpleTableReplicationState,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[schema(nullable = true)]
+    pub table_sync_lag: Option<SlotLagMetricsResponse>,
+}
+
+/// Lag metrics reported for replication slots.
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
+pub struct SlotLagMetricsResponse {
+    /// Bytes between the current WAL location and the slot restart LSN.
+    #[schema(example = 1024)]
+    pub restart_lsn_bytes: i64,
+    /// Bytes between the current WAL location and the confirmed flush LSN.
+    #[schema(example = 2048)]
+    pub confirmed_flush_lsn_bytes: i64,
+    /// How many bytes of WAL are still safe to build up before the limit of the slot is reached.
+    #[schema(example = 8192)]
+    pub safe_wal_size_bytes: i64,
+    /// Write lag expressed in milliseconds.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[schema(example = 1500, nullable = true)]
+    pub write_lag: Option<i64>,
+    /// Flush lag expressed in milliseconds.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[schema(example = 1200, nullable = true)]
+    pub flush_lag: Option<i64>,
```

Comment:
Do we need the `flush_lag` and `write_lag` separately? Are they shown separately on UI? I think we set them both to the same value in ETL right?

Reply:
We are showing `flush_lag` and returning `write_lag` just to have more info for debugging purposes.